pkg/kubelet/apis/deviceplugin/v1alpha/api.proto

```diff
@@ -48,6 +50,11 @@ service DevicePlugin {
 	// Plugin can run device specific operations and instruct Kubelet
 	// of the steps to make the Device available in the container
 	rpc Allocate(AllocateRequest) returns (AllocateResponse) {}
+
+        // PreStart is called, if indicated by Device Plugin during registeration phase,
+        // before each container start. Device plugin can run device specific operations
+        // such as reseting the device before making devices available to the container
+	rpc PreStart(PreStartRequest) returns (Empty) {}
```

Comment:
We are adding a new rpc call in this PR. That means any existing device plugins need to implement this rpc function when they sync up their api package. For this, should we create a v1beta1 directory and start to make changes there? I have proposed to discuss this in next week's rwmg meeting, but as you are working on this change, would like to raise this question again. Or maybe it is easier that I create a separate PR to just copy the current v1apha to v1beta1 and send it for review by some folks.

Reply:
@jiayingz Please not that though this PR is introducing a new RPC, still changes are backward compatible and existing device plugins will not break even after syncing with new DP api.

Reply:
I second @jiayingz's suggestion, given that grpc recommends updating versions for changes to service and protos.
But do keep in mind that updating versions won't be possible after GA and I suspect we may continue to add functionality.
So we should identify a way to maintain backwards compatibility while adding new features.
This may mean for example, kubelet not invoking this new `PreStart` RPC unless the server supports  it (discovered using grpc).